checker-qual/src/main/java/org/checkerframework/framework/qual/TargetLocations.java

```diff
@@ -7,31 +7,38 @@
 import java.lang.annotation.Target;
 
 /**
- * NOTE: This meta-annotation is <a
- * href="https://github.com/typetools/checker-framework/issues/1919"><b>not currently
- * enforced</b></a>.
- *
- * <p>A meta-annotation that restricts the type-use locations where a type qualifier may be written.
+ * A meta-annotation that restricts the type-use locations where a type qualifier may be applied.
  * When written together with {@code @Target({ElementType.TYPE_USE})}, the given type qualifier may
- * be written only at locations listed in the {@code @TargetLocations(...)} meta-annotation.
+ * be applied only at locations listed in the {@code @TargetLocations(...)} meta-annotation.
  * {@code @Target({ElementType.TYPE_USE})} together with no {@code @TargetLocations(...)} means that
- * the qualifier can be written on any type use.
+ * the qualifier can be applied on any type use. {@code @TargetLocations({})} will prevent the
+ * annotation from been applied in the source code, but you can achieve the same goal by writing
+ * {@code @Target({})}. So, we don't suggest writing the annotation with no type-use location
+ * supplied.
  *
- * <p>This enables a type system designer to permit a qualifier to be written only in certain
+ * <p>This enables a type system designer to permit a qualifier to be applied only in certain
  * locations. For example, some type systems' top and bottom qualifier (such as {@link
- * org.checkerframework.checker.nullness.qual.KeyForBottom}) should only be written on an explicit
+ * org.checkerframework.checker.regex.qual.RegexBottom}) should only be written on an explicit
  * wildcard upper or lower bound. This meta-annotation is a declarative, coarse-grained approach to
  * enable that. For finer-grained control, override {@code visit*} methods that visit trees in
  * BaseTypeVisitor.
  *
- * <p>This annotation does not prevent the type system from defaulting, inferring, or computing the
- * given type annotation at the given location. It only prevents users from writing an explicit
- * annotation at the given location.
+ * <p>This meta-annotation prevents the type system from inferring, or computing the given type
+ * annotation at the given location. It also prevents users from writing an explicit annotation at
+ * the given location.
```

Comment:
```suggestion
 * <p>This meta-annotation prevents the type system from inferring, or computing the given
 * qualifier at the given location. It also prevents users from writing that qualifier explicitly at
 * the given location.
```

Make sure to use `qualifier` for the type system... qualifiers. And `annotation` for the meta-annotations.